### PR TITLE
Feature UI

### PR DESCRIPTION
**Pull request by netanelc305**

### CODE REVIEW

File: hilda/hilda_client.py [hunk 1]
```diff
@@ -120,16 +122,22 @@ def lsof(self) -> dict:
         return {int(k): v for k, v in result.items()}
 
     @command()
-    def bt(self):
+    def bt(self, should_print=True, depth=-1):
```
Comment (doronz88) [inline hilda/hilda_client.py:125]: add return type

File: hilda/ui/views.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        for value in registers_set:
+            if self.rtype.lower() in value.GetName().lower():
+                return value
+        return None
+
+
+class DisassemblyView(View):
+    """ Implements disassembly view """
+
+    def __init__(self, hilda_client, color_scheme, num_of_inst=5):
```
Comment (doronz88) [inline hilda/ui/views.py:147]: Use class variable: `DEFAULT_INSTRUCTION_COUNT = 5`

File: hilda/ui/views.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        """
+        :param hilda_client: hilda.hilda_client.HildaClient
+        :param num_of_inst: int
+        :param color_scheme: hilda.ui.ui_manager.ColorScheme
+        """
+        super().__init__(hilda_client, 'Disassembly', color_scheme)
+        self.num_of_inst = num_of_inst
+
+    def __str__(self) -> str:
+        """ Format disassembly view for printing"""
```
Comment (doronz88) [inline hilda/ui/views.py:157]: Missing space

File: hilda/ui/views.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            if load_addr == self.hilda.frame.pc:
+                fmt += f'{"<-- $pc":^50}'
+            fmt_parts.append(fmt)
+        return super().__str__() + '\n'.join(fmt_parts)
+
+
+class BackTraceView(View):
+    """ Implements backtrace view """
+
+    def __init__(self, hilda_client, color_scheme, depth=5):
```
Comment (doronz88) [inline hilda/ui/views.py:177]: Use class variable: `DEFAULT_BACKTRACE_DEPTH = 5`

**Review by doronz88 — CHANGES_REQUESTED** [review]

File: hilda/hilda_client.py [hunk 5]
```diff
@@ -81,6 +82,7 @@ def __init__(self, debugger: lldb.SBDebugger):
         self.captured_objects = {}
         self.registers = Registers(self)
         self.arch = self.target.GetTriple().split('-')[0]
+        self.ui_manager = UiManager(self)
```
Comment (doronz88) [inline hilda/hilda_client.py:85]: add it as a global and explain how to use it in README

**Review by doronz88 — CHANGES_REQUESTED** [review]

File: hilda/ui/views.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        :param color_scheme: hilda.ui.ui_manager.ColorScheme
+        """
+        self.title = title
+        self.hilda = hilda_client
+        self.active = True
+        self.color_scheme = color_scheme
+
+    @abstractmethod
+    def __str__(self) -> str:
+        terminal_size = shutil.get_terminal_size()
```
[the remark this answers is not in the document]
Comment (netanelc305) [reply hilda/ui/views.py:36]: ?

File: hilda/ui/views.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+    def __init__(self, hilda_client, color_scheme, size=10):
+        """
+        :param hilda_client: hilda.hilda_client.HildaClient
+        :param size: int
+        :param color_scheme: hilda.ui.ui_manager.ColorScheme
+        """
+        super().__init__(hilda_client, 'Stack', color_scheme)
+        self.size = size
+        self.prev = None
```
[the remark this answers is not in the document]
Comment (netanelc305) [reply hilda/ui/views.py:53]: ?

File: hilda/ui/views.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+                  f'{style(hex(offset), fg=self.color_scheme.stack_offset)}: ' \
+                  f'{style(v, fg=data_color)}'
+            if offset == 0:
+                fmt += f'{"<-- $sp":^50}'
+            offset += 8
+            fmt_parts.append(fmt)
+
+        return super().__str__() + '\n'.join(fmt_parts)
+
+    def _create_mapping(self) -> Mapping:
```
[the remark this answers is not in the document]
Comment (netanelc305) [reply hilda/ui/views.py:74]: ?

**Review by doronz88 — CHANGES_REQUESTED** [review]

**Pull request merged**